apps/frontend/src/pages/project/components/setup-analytics-modal/setup-analytics-modal.tsx

```diff
@@ -0,0 +1,66 @@
+import { Button, Modal } from "~/libs/components/components.js";
+import { DataStatus } from "~/libs/enums/enums.js";
+import {
+	useAppDispatch,
+	useAppSelector,
+	useCallback,
+} from "~/libs/hooks/hooks.js";
+import { type UserAuthResponseDto } from "~/modules/auth/auth.js";
+import { actions as projectApiKeyActions } from "~/modules/project-api-keys/project-api-keys.js";
+import { type ProjectGetAllItemResponseDto } from "~/modules/projects/projects.js";
+
+import { Output } from "../components.js";
+import styles from "./styles.module.css";
+
+type Properties = {
+	currentUser: UserAuthResponseDto;
+	isOpened: boolean;
+	onClose: () => void;
+	project: ProjectGetAllItemResponseDto;
+};
+
+const SetupAnalyticsModal = ({
+	currentUser,
+	isOpened,
+	onClose,
+	project,
+}: Properties): JSX.Element => {
+	const dispatch = useAppDispatch();
+	const { dataStatus } = useAppSelector(({ projectApiKeys }) => projectApiKeys);
+
+	const hasProjectApiKey = project.apiKey !== null;
+	const isGenerateButtonDisabled =
+		hasProjectApiKey || dataStatus === DataStatus.PENDING;
+
+	const handleGenerateSubmit = useCallback(() => {
+		void dispatch(
+			projectApiKeyActions.create({
+				projectId: project.id,
+				userId: currentUser.id,
+			}),
+		);
+	}, [dispatch, project, currentUser]);
```

Comment:
I was thinking of creating implicit form, with default values, and no Input components to change the values. But do it this way for now. 
Should I adjust this by using `useAppForm` for submittion purpose?

Reply:
yes, we need to handle it using useAppForm